src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlServiceImpl.java

```diff
@@ -130,44 +152,123 @@ private String convertToValidCountryOfOrigin(String countryCode) {
+    private boolean shouldProccessAddress(JaggaerUploadSuppliersProcessingMode processingMode,
+            VendorAddress vendorAddress) {
+        return StringUtils.equals(processingMode.modeCode, JaggaerUploadSuppliersProcessingMode.VENDOR.modeCode) || vendorAddress.isActive();
+    }
+    
+    private PrimaryAddressList buildPrimaryAddressList(VendorDetail detail) {
+        PrimaryAddressList addressList = new PrimaryAddressList();
+        if (detail.isActiveIndicator()) {
+            for (VendorAddress vendorAddress : detail.getVendorAddresses()) {
+                JaggaerAddressTypeForXml addressType = JaggaerAddressTypeForXml.findJaggaerAddressTypeForXmlByKfsAddressType(vendorAddress.getVendorAddressTypeCode());
+                if (vendorAddress.isVendorDefaultAddressIndicator() && addressType != JaggaerAddressTypeForXml.PHYSICAL) {
+                    AssociatedAddress associatedAddress = new AssociatedAddress();
+                    associatedAddress.setType(addressType.jaggaerAddressType);
+                    
+                    AddressRef ref = new AddressRef();
+                    ref.setErpNumber(JaggaerBuilder.buildErpNumber(String.valueOf(vendorAddress.getVendorAddressGeneratedIdentifier())));
+                    associatedAddress.setAddressRef(ref);
+                    
+                    addressList.getAssociatedAddresses().add(associatedAddress);
+                }
+            }
+        }
+        return addressList;
+    }
+    
+    private LocationList buildLocationList(VendorDetail detail, JaggaerUploadSuppliersProcessingMode processingMode) {
+        LocationList locationList = new LocationList();
+        for (VendorAddress vendorAddress : detail.getVendorAddresses()) {
+            if (shouldProccessAddress(processingMode, vendorAddress)) {
+                Location location = new Location();
+                location.setErpNumber(JaggaerBuilder.buildErpNumber(String.valueOf(vendorAddress.getVendorAddressGeneratedIdentifier())));
+                location.setName(JaggaerBuilder.buildName(buildNameStringFromAddress(vendorAddress)));
+                
+                boolean locationActive = detail.isActiveIndicator() && vendorAddress.isActive();
+                location.setActive(JaggaerBuilder.buildActive(locationActive, JaggaerBooleanToStringTyoe.LOCATION_ACTIVE));
+                location.setLocationActive(JaggaerBuilder.buildJaggaerBasicValue(locationActive, JaggaerBooleanToStringTyoe.LOCATION_ACTIVE));
+                location.setPrimary(JaggaerBuilder.buildJaggaerBasicValue(locationActive && vendorAddress.isVendorDefaultAddressIndicator(), 
+                        JaggaerBooleanToStringTyoe.LOCATION_PRIMARY));
+                
+                locationList.getLocations().add(location);
+            }
+        }
+        return locationList;
+    }
+
+    private String buildNameStringFromAddress(VendorAddress vendorAddress) {
```

Comment:
When I saw this method, my first thought was "what name"?
Would buildLocationNameStringFromAddress make this clearer?

Reply:
The name is used more than for location I think, I will see if I find a better name